_posts/2021-01-12-github-application-running-on-faasd-behind-inlets.md

```diff
@@ -0,0 +1,508 @@
+$ faas-cli template store pull golang-middleware
+# Then, create the function itself.
+$ faas-cli new issues-bot --lang golang-middleware --prefix <DOCKER_HUB_ID>
+```
+After created the function, we need to define some arguments, environments and secrets for the function.
+Let's add them:
+
+```yaml
+  build_args:
+      GO111MODULE: on
+    secrets:
+      - webhook-secret # your secret goes here
+      - private-key-secret # your private key goes here
+    environment:
+      APP_ID: "" #your app id goes here
+```
+
+Finally, we need to create those secrets above with make use of faas-cli.
+
+Let's create our secrets
+
+```bash
+$ export WEBHOOK_SECRET="sup3rs3cr3t"
+$ faas-cli secret create webhook-secret --from-literal $WEBHOOK_SECRET
+# Download the private key to your host
+$ faas-cli secret create private-key-secret --from-file <path_to_your_pem_file>.pem
+```
+
+We should create a secret in faasd, in order to do that we need to access the Gateway of faasd
+```bash
+$ export IP=$(multipass info faasd --format json| jq '.info.faasd.ipv4[0]' | tr -d '\"')
+# Let's capture the authentication password into a file for use with faas-cli
+$ ssh ubuntu@$IP "sudo cat /var/lib/faasd/secrets/basic-auth-password" > basic-auth-password
+# Login from your laptop (the host)
+$ export OPENFAAS_URL=http://$IP:8080 && \
+cat basic-auth-password | faas-cli login -s
+```
+
+## Connect your client to the inlets-pro server using inletsctl
+We need to establish connection between our client, and the inlets-pro server in order to get events from there
+
+```bash
+$ export UPSTREAM=$IP # faasd gateway ip, we have already grap the URL above
+$ export PORTS=8080 # faasd gateway port
+$ export LICENSE="eyJhbGciOiJFUzI..."
+
+# Notice that this command is the output of the "inletsctl create" command above
+$ inlets-pro client --url "wss://XX.XXX.XXX.XX:8123/connect" \
+        --token "$TOKEN" \
+        --license "$LICENSE" \
+        --upstream $UPSTREAM \
+        --ports $PORTS
+```
+## Test
+In order to test it we need to install this app to selected repositories. Create a repository called "test-issues-bot", then install this app for it.
+![repository-access](/images/2021-01-12-github-application-using-go-and-inlets-pro/repository-access.png)
+
+Then, create an issue for the repository. You will see the message 
+> "Hello, issue opened by: developer-guy"
+
+![test-issue-bot](/images/2021-01-12-github-application-using-go-and-inlets-pro/test-issue-bot.png)
+
+Finally , let's close the issue by typing command */close*
+![close-issue](/images/2021-01-12-github-application-using-go-and-inlets-pro/close-issue.png)
+
+## Cleanup
+
+```bash
+$ multipass delete faasd
+$ multipass purge
+$ inletsctl delete --provider ec2 --id "YOUR_INSTANCE_ID" --access-token $AWS_ACCESS_KEY --secret-key $AWS_SECRET_KEY --region eu-central-1
+```
+
+## Further More
+You can run this Bot on a Raspberry Pi, with faasd and TLS with inlets-pro and Caddy!
```

Comment:
Whilst I like this idea, I feel it's a separate blog post, can you save this whole section somewhere and delete it from the blog post?

Reply:
done